### PR TITLE
New Sand Planet Ruin: Alien Facility

### PR DESCRIPTION
**Pull request by Draggeru**
## About The Pull Request

Adds a completely new ruin to the game, an alien facility on sand planets.
SDNM View:

In Game View:
ruin 1

My intent was to redo a map in the deprecated files, but the one I picked had abductor stuff. Which abductors no longer exist in the lore, so I swapped it out for xenos instead.

## Why It's Good For The Game

New ruins = good. New ruins with more than 2 braincells worth of effort = good. New Ruins also means I can bribe the maptainers into letting me PR a ship. :)

## Changelog

:cl:
add: Alien Facility ruin for sand planets
/:cl:

### CODE REVIEW

**Comment by meemofcourse** [comment]
> del: At least half of my braincells

don't keep this in ok thanks

**Comment by Draggeru** [comment]
> > del: At least half of my braincells
> 
> don't keep this in ok thanks

:(

**Review by spockye — CHANGES_REQUESTED** [review]
image

missing cable

image

Resin walls dont count as turf walls. you need to put floors under them!

image

please try to include some premade external areas around the two opening points. Pregen might cover them up fully, either making the exhaust, or entrance not end up well, functioning as intended. 

image

unconnected scrubber

image

please make your id's a bit more unique. include the name of the ruin or something!
helps with bugtesting later on

**Comment by Draggeru** [comment]
> image
> 
> missing cable
> 
> image
> 
> Resin walls dont count as turf walls. you need to put floors under them!
> 
> image
> 
> please try to include some premade external areas around the two opening points. Pregen might cover them up fully, either making the exhaust, or entrance not end up well, functioning as intended.
> 
> image
> 
> unconnected scrubber
> 
> image
> 
> please make your id's a bit more unique. include the name of the ruin or something! helps with bugtesting later on


Done

**Review by thgvr — CHANGES_REQUESTED** [review]
Remove the shrapnel mines.

**Comment by Draggeru** [comment]
I have no idea why the integration test is failing

**Comment by immortalartisan** [comment]
Check the details on the integration test seems to be something related to rad types. is there a rad hazard element on this map?

**Comment by MarkSuckerberg** [comment]
Sorry, it's a random fail that's fixed in #2364, I'll just re-run them for you

**Comment by Draggeru** [comment]
Oh I think I forgot a couple mines, don't TM this until I can get home to double check

**Comment by Draggeru** [comment]
We're good now, I forgor about the mine at the top. All good for a test merge again.

**Review by thgvr — APPROVED** [review]

**Review by goober3 — CHANGES_REQUESTED** [review]
please add more exterior! even if it's just, like. an empty field -- symmetrical lines of terrain will look odd, especially if this ruin spawns inside a mountain.

**Comment by Draggeru** [comment]
> please add more exterior! even if it's just, like. an empty field -- symmetrical lines of terrain will look odd, especially if this ruin spawns inside a mountain.



Done

**Comment by spockye** [comment]
do you intend for those sandbags to be deployed? if that case use the sturcture version. not the Item

**Comment by Draggeru** [comment]
> do you intend for those sandbags to be deployed? if that case use the sturcture version. not the Item

Oh, nah. I think I'll just axe em

**Review by Latentish — CHANGES_REQUESTED** [review]
You need to make a map catalogue entry.

File: config/sandruinblacklist.txt [hunk 1]
```diff
@@ -4,6 +4,7 @@
 #SPECIFYING AN INVALID MAP WILL RESULT IN RUNTIMES ON GAME START
 
 ##MISC
+#_maps/RandomRuins/SandRuins/whitesands_alien_facility.dmm
```
Comment (Latentish) [inline config/sandruinblacklist.txt]: This is a deprecated file that we do not upkeep. You should not make changes to it.

**Review by Latentish — CHANGES_REQUESTED** [review]

**Comment by Draggeru** [comment]
> You need to make a map catalogue entry.

Where is that in the code?

**Comment by Latentish** [comment]
> Where is that in the code?

You updated a file called the sandruinblacklist.txt. It's listed in the changes you made. As for the mapcatalogue, it's a txt in the maps folder. There's another txt that's a key for adding new entries.

**Review by meemofcourse — CHANGES_REQUESTED** [review]
is this thing on? ok cool

i have a couple of issues with this ruin:
* the hive is very symmetrical and organized. i would not imagine a xenomorph hive as a giant cube
* there is a lot of engineering gear in what should be a medical ruin, notably the inclusion of an atmospherics setup, multiple canisters and generators, making the ruin be a strange atmos-chemical-medical mix. change this!

**Comment by meemofcourse** [comment]
some other minor stuff:
* for what i imagine is a ruin raided by Unspecified Syndicate Force, it doesn't look very damaged. add some casings, broken glass, grilles and unsecured girders
* maybe try and make it less of a Square

**Review by Erikafox — CHANGES_REQUESTED** [review]
image
All of your exterior plating are going to be active turfs because they start as airless.

image
Don't do this
Do this
StrongDMM_uETgYtbZlJ

image
Put dir on button

image
Turn this into a proper laundering / cleaning area. It's spartan and the maint stuff sitting around hurts the image.

image
Lockers should be on +-28
StrongDMM_WA31ACVW5E
no virology content
image
No brainrot please clean it *all* out

image
This is a lot of food for a base that's in the middle of Shit Being Fucked. I would have some empty donk boxes and cut out an ingredient box.
image
This is very boxy. Please make it more organic / overgrown. It should seem like a hive. Intentional, but in an alien way
image
All turfs on the inside ring should have a new subtype made with planetary atmos disabled (to show that they're closed air)
image
Make this into a proper little shitty engineering locker room.
image
It would be cool if the threat was telegraphed by having the aliens starting to grow out of / against one of the walls
Needs a proper office (not a lobby), perhaps with some fluff reports and a fax machine.
Complete absence of light switches, the more civilian areas could use them.
image
What is the story supposed to be with this plasma anyways. Is someone supposed to rush in and ignite it? Set up a vent that's been clearly jammed up to show that it's a countermeasure that's been circumvented.

**Comment by Draggeru** [comment]
I've lost motivation to keep contributing to shiptest at current. Maybe I will revive this at some point down the line.

**Pull request closed without merging**